policy-controller/k8s/status/src/index.rs

```
@@ -0,0 +1,246 @@
+
+    async fn process_server_update(&mut self, patch_params: k8s::PatchParams) {
+        let route_ids: Vec<ResourceId> = self.index.read().http_routes.keys().cloned().collect();
+        for route_id in route_ids {
+            self.process_http_route_update(route_id, patch_params.clone())
+                .await;
+        }
+    }
+}
+
+impl Index {
+    pub fn shared(updates: UnboundedSender<Update>) -> SharedIndex {
+        Arc::new(RwLock::new(Self {
+            updates,
+            http_routes: HashMap::new(),
+            servers: HashSet::new(),
+        }))
+    }
+
+    fn update_http_route(&mut self, route_id: ResourceId, route_binding: RouteBinding) -> bool {
+        match self.http_routes.entry(route_id) {
+            Entry::Vacant(entry) => {
+                entry.insert(route_binding);
+            }
+            Entry::Occupied(mut entry) => {
+                if *entry.get() == route_binding {
+                    return false;
+                }
+                entry.insert(route_binding);
+            }
+        }
+        true
+    }
+}
+
+impl kubert::index::IndexNamespacedResource<k8s::policy::HttpRoute> for Index {
+    fn apply(&mut self, resource: k8s::policy::HttpRoute) {
+        let namespace = resource
+            .namespace()
+            .expect("HTTPRoute must have a namespace");
+        let name = resource.name_unchecked();
+        let id = ResourceId::new(namespace.clone(), name.clone());
+
+        let route_binding = match RouteBinding::try_from(resource) {
+            Ok(binding) => binding,
+            Err(error) => {
+                tracing::info!(%namespace, %name, %error, "Ignoring HTTPRoute");
+                return;
+            }
+        };
+
+        if self.update_http_route(id.clone(), route_binding) {
+            if let Err(error) = self.updates.send(Update::HttpRoute(id.clone())) {
+                tracing::error!(%id.namespace, %id.name, %error, "Failed to send HTTPRoute update")
+            }
```

I wonder if we can `map_err` here instead

`updates.send(..).map_err(|error| error!(..))`

I think Clippy will complain that the error is not handled though...